lib/foreman_tasks_core/task_launcher/group.rb

```diff
@@ -0,0 +1,122 @@
+require 'foreman_tasks_core/runner'
+
+module ForemanTasksCore
+  module TaskLauncher
+    class AbstractGroup < Batch
+      class GroupParentAction < ForemanTasksCore::TaskLauncher::ParentAction
+        def plan(launcher, input_hash)
+          launcher.launch_children(self, input_hash)
+          sequence do
+            results = plan_self
+            plan_action BatchCallback, launcher.prepare_batch(input_hash), results.output[:results]
+          end
+        end
+
+        def run(event = nil)
+          super unless event == Dynflow::Action::Skip
+        end
+
+        def check_for_errors!(optional = true)
+          super unless optional
+        end
+
+        def on_finish
+          output[:results] = sub_plans.map(&:entry_action).reduce({}) do |acc, cur|
+            acc.merge(cur.execution_plan_id => cur.output)
+          end
+        end
+
+        def finalize
+          output.delete(:results)
+          check_for_errors!
+        end
+
+        def initiate
+          ping suspended_action
+          wait_for_sub_plans sub_plans
+        end
+
+        def rescue_strategy_for_self
+          Dynflow::Action::Rescue::Skip
+        end
+      end
+
+      class BatchCallback < ::Dynflow::Action
+        def plan(input_hash, results)
+          plan_self :targets => input_hash, :results => results
+        end
+
+        def run
+          payload = format_payload(input['targets'], input['results'])
+          SmartProxyDynflowCore::Callback::Request.new.callback({ :callbacks => payload }.to_json)
+        end
+
+        private
+
+        def format_payload(input_hash, results)
+          input_hash.map do |task_id, callback|
+            { :callback => callback, :data => results[task_id] }
+          end
+        end
+      end
+
+      class Dummy < ::ForemanTasksCore::Runner::Action
```

Comment:
Naming is hard, but `Dummy` is quite confusing when looking at the dynflow console. What about calling it `ChildAction`, and `GroupParentAction` would be just `ParentAction`